### PR TITLE
add dsl.Collected docs

### PR DESCRIPTION
**Pull request by connor-mccarthy**

### CODE REVIEW

File: content/en/docs/components/pipelines/v2/author-a-pipeline/pipelines.md [hunk 1]
```diff
@@ -68,6 +68,55 @@ def my_pipeline():
...
+def my_pipeline():
+    with dsl.ParallelFor(
+        items=[1, 5, 10, 25],
+        parallelism=2
+    ) as epochs:
+        train_model_task = train_model(epochs=epochs)
+    max_accuracy(models=dsl.Collected(train_model_task.outputs['model']))
+```
+
+Downstream tasks may consume `dsl.Collected` outputs via an input annotated with a `List` of parameters or `List` of artifacts. For example, `max_accuracy` in the above example has the input `models` with type `List[Model]`:
```
Comment (sen-sam) [inline content/en/docs/components/pipelines/v2/author-a-pipeline/pipelines.md:88]: Suggestions:
- Replace "may" with "might".
- Add "a" after "or".
- Replace "above" with "preceding". Our style recommendation generally discourages using "above" to denote position in a document.
- Is it possible to have a sentence describing the next code sample?

File: content/en/docs/components/pipelines/v2/author-a-pipeline/pipelines.md [hunk 2]
```diff
@@ -68,6 +68,55 @@ def my_pipeline():
...
+```python
+from kfp import dsl
+from kfp.dsl import Model, Output
+
+@dsl.component
+def select_best(models: List[Model]) -> float:
+    return max(score_model(model) for model in models)
+```
+
+`dsl.Collected` can be used to collect outputs from nested loops as well.
```
Comment (sen-sam) [inline content/en/docs/components/pipelines/v2/author-a-pipeline/pipelines.md:99]: Rephrase: "You can also use ..."

File: content/en/docs/components/pipelines/v2/author-a-pipeline/pipelines.md [hunk 3]
```diff
@@ -68,6 +68,55 @@ def my_pipeline():
...
+
+Collection of *parameters* from nested loops will result in a *nested list* of parameters (e.g., two nested `dsl.ParallelFor` groups results in a **list of lists** of parameters). By comparison, collection of *artifacts* from nested loops results in a *flattened* lists of artifacts (e.g., any number of nested `dsl.ParallelFor` groups results in a **single list** containing all artifacts).
```
Comment (sen-sam) [inline content/en/docs/components/pipelines/v2/author-a-pipeline/pipelines.md:101]: Is it possible to combine this with the previous paragraph?

Suggest replacing "e.g." with "For example, ..." starting a new sentence and removing the parentheses.

If formatting keywords like "parameters" is necessary, I recommend using the same formatting for these e.g. italics for "parameters", "flattened", and "single list".

File: content/en/docs/components/pipelines/v2/author-a-pipeline/pipelines.md [hunk 4]
```diff
@@ -68,6 +68,55 @@ def my_pipeline():
...
+
+You may also return a `dsl.Collected` from a pipeline using a `List` of parameters or `List` of artifacts return annotation:
```
Comment (sen-sam) [inline content/en/docs/components/pipelines/v2/author-a-pipeline/pipelines.md:103]: Similar comment as above - replace "may" with "might", add "a" after "or".

Is "artifacts return annotation" a single term?

**Review by sen-sam — CHANGES_REQUESTED** [review]

File: content/en/docs/components/pipelines/v2/author-a-pipeline/pipelines.md (hunk 1 above)
Comment (connor-mccarthy) [reply content/en/docs/components/pipelines/v2/author-a-pipeline/pipelines.md:88]: Updated.

File: content/en/docs/components/pipelines/v2/author-a-pipeline/pipelines.md (hunk 2 above)
Comment (connor-mccarthy) [reply content/en/docs/components/pipelines/v2/author-a-pipeline/pipelines.md:99]: Updated.

File: content/en/docs/components/pipelines/v2/author-a-pipeline/pipelines.md (hunk 4 above)
Comment (connor-mccarthy) [reply content/en/docs/components/pipelines/v2/author-a-pipeline/pipelines.md:103]: Updated.

It's not really a single term. "`List` of artifacts" is one thing and the "return annotation" is how it's used. I updated this slightly for clarity.

File: content/en/docs/components/pipelines/v2/author-a-pipeline/pipelines.md [hunk 5]
```diff
@@ -68,6 +68,57 @@ def my_pipeline():
...
+    return max(score_model(model) for model in models)
+```
+
+You can also use `dsl.Collected` to collect outputs from nested loops as well.
+
+Collection of *parameters* from nested loops will result in a *nested list* of parameters. For example, collection of parameters from two nested `dsl.ParallelFor` groups results in a *list of lists* of parameters.
+
+By comparison, collection of *artifacts* from nested loops results in a *flattened* lists of artifacts. For example, collection of artifacts from five nested `dsl.ParallelFor` groups still results in a *single list* containing all artifacts.
+
+You might also return a `dsl.Collected` from a pipeline using a return annotation of a `List` of parameters or a `List` of artifacts:
```
[the remark this answers is not in the document]
Comment (sen-sam) [reply content/en/docs/components/pipelines/v2/author-a-pipeline/pipelines.md:105]: "You can also return a `dsl.Collected` from a pipeline. Use a `List` of parameters or a `List` of artifacts in the return annotation, as shown in the following example:"

**Review by sen-sam — CHANGES_REQUESTED** [review]
A few nits here and there.

File: content/en/docs/components/pipelines/v2/author-a-pipeline/pipelines.md [hunk 6]
```diff
@@ -68,6 +68,57 @@ def my_pipeline():
...
+def my_pipeline():
+    with dsl.ParallelFor(
+        items=[1, 5, 10, 25],
+        parallelism=2
+    ) as epochs:
+        train_model_task = train_model(epochs=epochs)
+    max_accuracy(models=dsl.Collected(train_model_task.outputs['model']))
+```
+
+Downstream tasks might consume `dsl.Collected` outputs via an input annotated with a `List` of parameters or a `List` of artifacts. For example, `max_accuracy` in the preceding example has the input `models` with type `Output[List[Model]]`, shown by the following component definition:
```
[the remark this answers is not in the document]
Comment (connor-mccarthy) [reply content/en/docs/components/pipelines/v2/author-a-pipeline/pipelines.md:88]: Done.

File: content/en/docs/components/pipelines/v2/author-a-pipeline/pipelines.md [hunk 7]
```diff
@@ -68,6 +68,57 @@ def my_pipeline():
...
+```python
+from kfp import dsl
+from kfp.dsl import Model, Output
+
+@dsl.component
+def select_best(models: Output[List[Model]]) -> float:
+    return max(score_model(model) for model in models)
+```
+
+You can also use `dsl.Collected` to collect outputs from nested loops as well.
```
[the remark this answers is not in the document]
Comment (connor-mccarthy) [reply content/en/docs/components/pipelines/v2/author-a-pipeline/pipelines.md:99]: Done.

File: content/en/docs/components/pipelines/v2/author-a-pipeline/pipelines.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (connor-mccarthy) [reply content/en/docs/components/pipelines/v2/author-a-pipeline/pipelines.md:105]: Done.

File: content/en/docs/components/pipelines/v2/author-a-pipeline/pipelines.md [hunk 8]
```diff
@@ -68,6 +68,57 @@ def my_pipeline():
...
+
+Collection of *parameters* from nested loops will result in a *nested list* of parameters. For example, collection of parameters from two nested `dsl.ParallelFor` groups results in a *list of lists* of parameters.
+
+By comparison, collection of *artifacts* from nested loops results in a *flattened* lists of artifacts. For example, collection of artifacts from five nested `dsl.ParallelFor` groups still results in a *single list* containing all artifacts.
```
[the remark this answers is not in the document]
Comment (connor-mccarthy) [reply content/en/docs/components/pipelines/v2/author-a-pipeline/pipelines.md]: Yes, flat list is correct. I've updated. WDYT?

File: content/en/docs/components/pipelines/v2/author-a-pipeline/pipelines.md [hunk 9]
```diff
@@ -68,6 +68,57 @@ def my_pipeline():
...
+```python
+from kfp import dsl
+from kfp.dsl import Model, Output
+
+@dsl.component
+def select_best(models: Output[List[Model]]) -> float:
+    return max(score_model(model) for model in models)
+```
+
+You can use `dsl.Collected` to collect outputs from nested loops.
```
Comment (sen-sam) [inline content/en/docs/components/pipelines/v2/author-a-pipeline/pipelines.md:99]: [Suggested rewording]

You can use `dsl.Collected` to collect outputs from nested loops in a *nested list* of parameters. For example, output parameters from two nested `dsl.ParallelFor` groups are collected in a multilevel nested list of parameters, where each nested list contains the output parameters from one of the `dsl.ParallelFor` groups. The number of nested levels is based on the depth of the `ParallelFor` group. \

By comparison, *artifacts* from nested loops are collected in a *flat* list.

Comment (connor-mccarthy) [reply content/en/docs/components/pipelines/v2/author-a-pipeline/pipelines.md]: Updated and merged with this suggestion.

**Pull request merged**